### PR TITLE
tests, signer: remove staticcheck warnings

### PR DESCRIPTION
**Pull request by gballet**
Taking over from @fjl, keeping them small to increase reviewing speed. Remove the following warnings:

```
signer/core/api.go:272:34: error strings should not be capitalized (ST1005)
signer/core/api.go:554:11: this value of err is never used (SA4006)
signer/core/api_test.go:74:14: don't use Yoda conditions (ST1017)
signer/core/api_test.go:94:5: don't use Yoda conditions (ST1017)
signer/core/cliui.go:63:26: func (*CommandlineUI).readPassword is unused (U1000)
signer/core/cliui.go:78:26: func (*CommandlineUI).readPasswordText is unused (U1000)
signer/core/stdioui.go:30:2: field mu is unused (U1000)
signer/core/types.go:63:20: error strings should not be capitalized (ST1005)
signer/core/uiapi.go:176:25: error strings should not be capitalized (ST1005)
signer/rules/rules.go:155:26: error strings should not be capitalized (ST1005)
signer/storage/storage.go:45:2: field namespace is unused (U1000)
signer/storage/storage.go:86:23: error strings should not be capitalized (ST1005)
tests/block_test_util.go:182:27: error strings should not be capitalized (ST1005)
tests/block_test_util.go:192:27: error strings should not be capitalized (ST1005)
tests/block_test_util.go:196:26: error strings should not be capitalized (ST1005)
tests/block_test_util.go:201:26: error strings should not be capitalized (ST1005)
tests/block_test_util.go:210:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:213:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:219:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:222:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:225:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:228:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:231:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:234:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:237:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:240:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:243:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:252:20: error strings should not be capitalized (ST1005)
tests/block_test_util.go:290:21: error strings should not be capitalized (ST1005)
tests/init_test.go:139:24: func (*testMatcher).whitelist is unused (U1000)
tests/transaction_test_util.go:89:22: error strings should not be capitalized (ST1005)
tests/transaction_test_util.go:95:21: error strings should not be capitalized (ST1005)
tests/transaction_test_util.go:101:21: error strings should not be capitalized (ST1005)
tests/transaction_test_util.go:107:21: error strings should not be capitalized (ST1005)
```

The two following errors are found in auto-generated code, so I will attempt to create a PR for the generator:

```
signer/rules/deps/bindata.go:23:25: error strings should not be capitalized (ST1005)
signer/rules/deps/bindata.go:31:25: error strings should not be capitalized (ST1005)
```

### CODE REVIEW

**Comment by gballet** [comment]
Turns out `go-bindata` has been fixed, the file just needed to be regenerated.

File: tests/init_test.go [hunk 1]
```diff
@@ -136,10 +136,6 @@ func (tm *testMatcher) fails(pattern string, reason string) {
 	tm.failpat = append(tm.failpat, testFailure{regexp.MustCompile(pattern), reason})
 }
 
-func (tm *testMatcher) whitelist(pattern string) {
```
Comment (holiman) [inline tests/init_test.go]: Hm, from time to time, I have to dig into the tests to find out why a particular one doesn't work, whenever the tests are updated. At those points in time, I heavily rely on this method -- it's  PITA to rerun every consensus test if I only want one. After it's all fixed, I remove that code again. 
So can we please leave it in, _and_ make the linter happy?

Comment (gballet) [reply tests/init_test.go]: I guess I can create a dummy test...

**Comment by fjl** [comment]
Please create one PR per package if possible. Maybe best to fix the `whitelist` thing by adding a test for the `whitelist` method itself

File: tests/init_test.go [hunk 2]
```diff
@@ -140,6 +140,11 @@ func (tm *testMatcher) whitelist(pattern string) {
 	tm.whitelistpat = regexp.MustCompile(pattern)
 }
 
+func TestDummyMatcherToKeepBothStaticcheckandHolimanHappy(t *testing.T) {
```
Comment (fjl) [inline tests/init_test.go]: This test name doesn't work for me.

Comment (gballet) [reply tests/init_test.go]: and what test name would work for you?

Comment (fjl) [reply tests/init_test.go]: `TestMatcherWhitelist` maybe?

Comment (gballet) [reply tests/init_test.go]: github didn't refresh, didn't see your suggestion :angry:

File: tests/init_test.go [hunk 3]
```diff
@@ -140,9 +140,15 @@ func (tm *testMatcher) whitelist(pattern string) {
 	tm.whitelistpat = regexp.MustCompile(pattern)
 }
 
-func TestDummyMatcherToKeepBothStaticcheckandHolimanHappy(t *testing.T) {
-	tm := testMatcher{}
-	tm.whitelist("")
+func TestWhitelist(t *testing.T) {
```
Comment (fjl) [inline tests/init_test.go]: Please move the test somewhere else in the file. While this is a test file, the code around here defines an API for use by other tests. Best to not dump a unit test right in the middle of that.

**Pull request merged**